ucp/_libs/ucx_api.pyx

```diff
@@ -495,6 +638,36 @@ cdef class UCXWorker(UCXObject):
         )
         context.add_child(self)
 
+    IF CY_UCP_AM_SUPPORTED:
```

Comment:
I suggest that you put `IF CY_UCP_AM_SUPPORTED:` inside `register_am_allocator()` and raise an error if AM isn't supported.

Reply:
Good idea, done in https://github.com/rapidsai/ucx-py/pull/707/commits/6ec6b447071f24de51593fba988d3dbca1ab33ff .